pandas/core/arrays/arrow/array.py

```diff
@@ -57,6 +57,76 @@
         "ge": pc.greater_equal,
     }
 
+    ARROW_LOGICAL_FUNCS = {
+        "and": NotImplemented if pa_version_under2p0 else pc.and_kleene,
+        "rand": NotImplemented
+        if pa_version_under2p0
+        else lambda x, y: pc.and_kleene(y, x),
+        "or": NotImplemented if pa_version_under2p0 else pc.or_kleene,
+        "ror": NotImplemented
+        if pa_version_under2p0
+        else lambda x, y: pc.or_kleene(y, x),
+        "xor": NotImplemented if pa_version_under2p0 else pc.xor,
+        "rxor": NotImplemented if pa_version_under2p0 else lambda x, y: pc.xor(y, x),
+    }
+
+    def cast_for_truediv(
+        arrow_array: pa.ChunkedArray, pa_object: pa.Array | pa.Scalar
+    ) -> pa.ChunkedArray:
+        # Ensure int / int -> float mirroring Python/Numpy behavior
+        # as pc.divide_checked(int, int) -> int
+        if pa.types.is_integer(arrow_array.type) and pa.types.is_integer(
+            pa_object.type
+        ):
+            return arrow_array.cast(pa.float64())
+        return arrow_array
+
+    def floordiv_compat(
+        left: pa.ChunkedArray | pa.Array | pa.Scalar,
+        right: pa.ChunkedArray | pa.Array | pa.Scalar,
+    ) -> pa.ChunkedArray:
+        # Ensure int // int -> int mirroring Python/Numpy behavior
+        # as pc.floor(pc.divide_checked(int, int)) -> float
+        result = pc.floor(pc.divide_checked(left, right))
+        if pa.types.is_integer(left.type) and pa.types.is_integer(right.type):
+            result = result.cast(left.type)
+        return result
+
+    ARROW_ARITHMETIC_FUNCS = {
+        "add": NotImplemented if pa_version_under2p0 else pc.add_checked,
+        "radd": NotImplemented
+        if pa_version_under2p0
+        else lambda x, y: pc.add_checked(y, x),
+        "sub": NotImplemented if pa_version_under2p0 else pc.subtract_checked,
+        "rsub": NotImplemented
+        if pa_version_under2p0
+        else lambda x, y: pc.subtract_checked(y, x),
+        "mul": NotImplemented if pa_version_under2p0 else pc.multiply_checked,
+        "rmul": NotImplemented
+        if pa_version_under2p0
+        else lambda x, y: pc.multiply_checked(y, x),
+        "truediv": NotImplemented
+        if pa_version_under2p0
+        else lambda x, y: pc.divide_checked(cast_for_truediv(x, y), y),
+        "rtruediv": NotImplemented
+        if pa_version_under2p0
+        else lambda x, y: pc.divide_checked(y, cast_for_truediv(x, y)),
+        "floordiv": NotImplemented
+        if pa_version_under2p0
+        else lambda x, y: floordiv_compat(x, y),
+        "rfloordiv": NotImplemented
+        if pa_version_under2p0
+        else lambda x, y: floordiv_compat(y, x),
+        "mod": NotImplemented,
+        "rmod": NotImplemented,
+        "divmod": NotImplemented,
+        "rdivmod": NotImplemented,
+        "pow": NotImplemented if pa_version_under2p0 else pc.power_checked,
```

Comment:
failing with pyarrow 2.0.0 

```python
ImportError while loading conftest '/home/simon/pandas/pandas/conftest.py'.
pandas/__init__.py:48: in <module>
    from pandas.core.api import (
pandas/core/api.py:27: in <module>
    from pandas.core.arrays import Categorical
pandas/core/arrays/__init__.py:20: in <module>
    from pandas.core.arrays.string_arrow import ArrowStringArray
pandas/core/arrays/string_arrow.py:37: in <module>
    from pandas.core.arrays.arrow import ArrowExtensionArray
pandas/core/arrays/arrow/__init__.py:1: in <module>
    from pandas.core.arrays.arrow.array import ArrowExtensionArray
pandas/core/arrays/arrow/array.py:124: in <module>
    "pow": NotImplemented if pa_version_under2p0 else pc.power_checked,
E   AttributeError: module 'pyarrow.compute' has no attribute 'power_checked'
```

Reply:
I merged @mroeschke's follow up PR (https://github.com/pandas-dev/pandas/pull/47752), but I am now wondering: we don't have CI builds with pyarrow 2.0 to catch this?

Reply:
We don't, only testing 1.01 (min), 5, 6, 7. IIRC pyarrow version 2 or 3 were causing the CI to time out around the read_csv tests so that's why 2 or 3 weren't included https://github.com/pandas-dev/pandas/issues/43650